phpBB/viewonline.php

```diff
@@ -470,20 +470,25 @@
 }
 $result = $db->sql_query($sql);
 
-$legend = '';
+$legend = [];
 while ($row = $db->sql_fetchrow($result))
 {
-	if ($row['group_name'] == 'BOTS')
+	$colour_text = ($row['group_colour']) ? ' style="color:#' . $row['group_colour'] . '"' : '';
+	$group_name = $group_helper->get_name($row['group_name']);
+
+	if ($row['group_name'] == 'BOTS' || ($user->data['user_id'] != ANONYMOUS && !$auth->acl_get('u_viewprofile')))
```

Comment:
The addition check doesn't make too much sense since we already check this at the top of viewonline:
https://github.com/phpbb/phpbb/blob/3.3.x/phpBB/viewonline.php#L36

Reply:
You are right.. I'll change it later. What about master branch?.. I'm sure it has same issue.